src/utilities/IOManager.cpp

```diff
@@ -10,128 +10,95 @@
-    return feedbackMap;
-}
+void IOManager::publishSettings(proto::Setting setting) { settingsPublisher->send(setting); }
 
-void IOManager::publishRobotCommand(proto::RobotCommand cmd, rtt::world::World const* world) {
-    if (!pause->getPause()) {
-        if (world->getWorld()) {
-            auto robot = world->getWorld()->getRobotForId(cmd.id(), true);
-            if (robot) {
-                if (cmd.kicker()) {
-                    interface::Input::drawData(interface::Visual::SHOTLINES, {robot->get()->getPos()}, Qt::green, robot->get()->getId(), interface::Drawing::CIRCLES, 36, 36, 8);
-                }
-                if (cmd.chip_kick_forced()) {
-                    interface::Input::drawData(interface::Visual::SHOTLINES, {robot->get()->getPos()}, Qt::green, robot->get()->getId(), interface::Drawing::DOTS, 36, 36, 8);
-                }
-                if (cmd.chipper()) {
-                    interface::Input::drawData(interface::Visual::SHOTLINES, {robot->get()->getPos()}, Qt::yellow, robot->get()->getId(), interface::Drawing::CIRCLES, 36, 36, 8);
-                }
-                // robot->setDribblerState(cmd.dribbler());
-            }
-        }
-        // sometimes trees are terminated without having a role assigned.
-        // It is then possible that a skill gets terminated with an empty robot: and then the id can be for example -1.
-        if (cmd.id() >= 0 && cmd.id() < 16) {
-            robotCommandPublisher->send(cmd);
+void IOManager::publishAllRobotCommands(const std::vector<proto::RobotCommand>& robotCommands) {
+    if(!pause->getPause()) {
+        proto::AICommand command;
+        for(const auto& robotCommand : robotCommands){
+          proto::RobotCommand * protoCommand = command.mutable_commands()->Add();
+          protoCommand->CopyFrom(robotCommand);
         }
+        command.mutable_extrapolatedworld()->CopyFrom(getState().command_extrapolated_world());
+        robotCommandPublisher->send(command);
     }
 }
-
-const proto::DemoRobot &IOManager::getDemoInfo() {
-    std::lock_guard<std::mutex> lock(demoMutex);
-    return this->demoInfoMsg;
+void IOManager::handleCentralServerConnection(){
+  //first receive any setting changes
+  bool received = true;
+  int numReceivedMessages = 0;
+  while(received){
+    auto receivedUIOptions = central_server_connection->read_next<proto::UiSettings>();
+    if (receivedUIOptions.is_ok()){
+      //TODO: process value
+      receivedUIOptions.value().PrintDebugString();
+      numReceivedMessages ++;
+    }else{
+      received = false;
+      //we don't print the errors as they mark there are no more messages
+    }
+  }
+  if(numReceivedMessages>0){
+    std::cout<<"received " << numReceivedMessages <<" packets from central server"<<std::endl;
+  }
+  //TODO: actually change settings at the relevant places within our AI
+  //then, send the current state once
+  //TODO: make sure to write/add relevant debug information/visualizations (strategy debug, etc.)
```

Comment:
todo?>

Reply:
Yes? I can't really do these unless it's clear what's supposed to be in the interface.